### PR TITLE
Enabling CRLF

### LINKED ISSUE #31: Support Windows line endings (crlf)
This came up when I tried to make metalsmith tests run on windows (https://github.com/metalsmith/metalsmith/issues/158). One of the reasons of failures was that gray-matter does not support Windows line endings.

### PR DESCRIPTION
**Pull request by moozzyk**
Fixes #31

### CODE REVIEW

**Comment by moozzyk** [comment]
Things I am not sure about:
 - is the coverage is enough
 - [stringify](https://github.com/jonschlinkert/gray-matter/blob/e948648d45e304f6bdbc0f93857ae74c94073062/index.js#L165-L173) uses hardcoded '\n' - maybe should use `require('os').EOL;`

File: .gitignore [hunk 1]
```diff
@@ -15,6 +15,7 @@ _test
 # OS or Editor folders
 .komodotools
 *.sublime-*
+.vscode/
```
Comment (jonschlinkert) [inline .gitignore]: some maintainers don't like this, I personally don't mind it a bit. Thanks for caring enough to do this.

File: index.js [hunk 2]
```diff
@@ -97,6 +97,8 @@ function matter(str, options) {
   var con = str.substr(end + b.length);
   if (con.charAt(0) === '\n') {
     con = con.substr(1);
+  } else if (con.charAt(0) === '\r' && con.charAt(1) === '\n') {
+    con = con.substr(2);
   }
```
Comment (jonschlinkert) [inline index.js]: Alternatively - if for some reason there are situations where this fails - we could just normalize the string to use all `\n`. but checking for carriage returns like this is probably much faster.

Comment (moozzyk) [inline index.js]: I tried to workaround this at the metalsmith level by normalizing new lines to `\n` but the principle in metalsmith is not to do changes like this on its own (https://github.com/metalsmith/metalsmith/pull/256#discussion_r84148370).

**Comment by moozzyk** [comment]
Re stringify - as you said Windows handles \n just fine and it was the reason I did not change this. I would leave it as is until this becomes a problem which is unlikely.

**Comment by jonschlinkert** [comment]
> I would leave it as is until this becomes a problem which is unlikely.

Ok that sounds fine to me, thanks for the feedback. I'll be back online tomorrow around 1 PM EST or so. I'll try to merge shortly thereafter. thanks!

**Comment by jonschlinkert** [comment]
Published and released as 2.1.0, thanks again @moozzyk! (this was arguably a bugfix, but I'd rather play it safe)


> FWIW, this looks good to me! 

thanks @Ajedi32!

**Pull request merged**